rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/actions/StartUnityAction.kt

```diff
@@ -21,29 +29,48 @@ open class StartUnityAction : DumbAwareAction() {
 
     override fun update(e: AnActionEvent) {
         val model = e.project?.solution?.frontendBackendModel
+        @NlsSafe
         val version = model?.unityApplicationData?.valueOrNull?.applicationVersion
 
         if (version != null)
-            e.presentation.text = "Start Unity ($version)"
+            e.presentation.text = UnityPluginActionsBundle.message("action.start.unity.text", version)
 
         e.presentation.isEnabled = version != null && !e.project.isConnectedToEditor()
         super.update(e)
     }
 
     companion object {
-        fun startUnity(project: Project, vararg args: String): Process? {
-            val processBuilderArgs = getUnityArgs(project).withProjectPath(project).withRiderPath()
-            processBuilderArgs.addAll(args)
-            return startUnity(processBuilderArgs)
+        private val logger = Logger.getInstance(StartUnityAction::class.java)
+        fun startUnity(project: Project): Process? {
+            val runManager = RunManager.getInstance(project)
+            val settings =
+                runManager.findConfigurationByTypeAndName(UnityExeConfigurationType.id, DefaultRunConfigurationGenerator.RUN_DEBUG_START_UNITY_CONFIGURATION_NAME)
+
+            if (settings != null){
+                val exeConfiguration = settings.configuration as UnityExeConfiguration
+                val runCommandLine = createEmptyConsoleCommandLine(exeConfiguration.parameters.useExternalConsole)
+                    .withEnvironment(exeConfiguration.parameters.envs)
+                    .withParentEnvironmentType(if (exeConfiguration.parameters.isPassParentEnvs) {
+                        GeneralCommandLine.ParentEnvironmentType.CONSOLE
+                    } else {
+                        GeneralCommandLine.ParentEnvironmentType.NONE
+                    })
+                    .withExePath(exeConfiguration.parameters.exePath)
+                    .withWorkDirectory(exeConfiguration.parameters.workingDirectory)
+                    .withRawParameters(exeConfiguration.parameters.programParameters)
+
+                return runCommandLine.toProcessBuilder().start()
```

Comment:
Is there a reason you start the process manually here instead of using something like `ExecutionUtil.runConfiguration`?

That way, some details like before-launch-actions would be taken into account.

Reply:
https://jetbrains.slack.com/archives/C53QFRZ7G/p1655808937561529